Source/WebCore/Modules/filesystemaccess/FileSystemSyncAccessHandle.cpp

```diff
@@ -158,4 +167,21 @@ void FileSystemSyncAccessHandle::invalidate()
     closeInternal(ShouldNotifyBackend::No);
 }
 
+bool FileSystemSyncAccessHandle::requestSpaceForWrite(uint64_t writeOffset, uint64_t writeLength)
+{
+    CheckedUint64 newSize = writeOffset;
+    newSize += writeLength;
+    if (newSize.hasOverflowed())
+        return false;
```

Comment:
Should we trigger an exception?

Reply:
The spec does not specify what to do when newSize is overflowed, but mentions if newSize - oldSize > quota, then we should throw quota error, so I think we should throw quota error here.